src/qt/walletmodel.cpp

```diff
@@ -527,7 +527,11 @@ WalletModel::SendCoinsReturn WalletModel::prepareTransaction(WalletModelTransact
             if ((total + nFeeRequired) > nBalance) {
                 return SendCoinsReturn(AmountWithFeeExceedsBalance);
             }
-            Q_EMIT message(tr("Send Coins"), tr("Transaction creation failed!\n%1").arg(QString::fromStdString(strFailReason)),
+
+            Q_EMIT message(tr("Send Coins"), tr("Transaction creation failed!\n%1").arg(
+                    strFailReason == "Transaction too large" ?
```

Comment:
Not for this PR but this line is, code wise, too forced. Would be good to have proper code error numbers (mapped to the string description) instead of return error strings and do this forced check + re translation here.

Reply:
Yes, for sure it would be a good improvement for the future, to use return codes (and rely on them) rather than fixed strings.